### PR TITLE
Spark Operator: Add monitoring resources

### PR DESCRIPTION
**Pull request by alembiewski**
This PR introduces support for automatic installation and configuration of monitoring resources for Spark Operator.

### CODE REVIEW

File: repository/spark/docs/beta1/monitoring.md [hunk 1]
```diff
@@ -16,77 +16,10 @@ If you don't already have the `prometheus-operator` installed on your Kubernetes
...
-   spec:
-     ports:
-       - port: 8090
-         name: metrics
-     clusterIP: None
-     selector:
-       "metrics-exposed": "true"
-   EOF
-   ```
+Full full list of metrics configuration parameters and defaults is available in KUDO Spark [params.yaml](../../operator/params.yaml).
```
Comment (akirillov) [inline repository/spark/docs/beta1/monitoring.md]: looks like a duplication here: `Full full list...`

File: repository/spark/operator/templates/spark-monitoring.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  name: spark-monitor
+  namespace: {{ .Namespace }}
+spec:
+  endpoints:
+    - interval: {{ .Params.metricsPollingInterval }}
+      port: metrics
+  selector:
+    matchLabels:
+      spark/servicemonitor: "true"
+{{- end }}
```
Comment (akirillov) [inline repository/spark/operator/templates/spark-monitoring.yaml]: looks like newline is missing here

File: repository/spark/docs/beta1/monitoring.md [hunk 3]
```diff
@@ -16,77 +16,10 @@ If you don't already have the `prometheus-operator` installed on your Kubernetes
...
-kubectl kudo install spark --instance=spark-operator \
-        -p enableMetrics=true \
-        -p operatorMetricsPort=10254 \
-        -p appMetricsPort=8090 \
-        -p metricsEndpoint=/metrics \
-        -p metricsPrefix="" \
-        -p metricsPollingInterval="5s"
-```
+Metrics reporting is enabled by default. Service endpoints and ServiceMonitor resources are configured to work with Prometheus Operator
+out of the box.
```
Comment (akirillov) [inline repository/spark/docs/beta1/monitoring.md]: I think it is worth mentioning somewhere in this section, that `appMetricsPort` should be the same for all submitted Spark Applications in order for metrics to be scraped

**Review by akirillov — COMMENTED** [review]
Thanks, @alembiewski. Overall LGTM, please check out the comments.

Also, these changes will not work on KUDO versions below 0.10.0 -  can you please update prerequisites?

File: repository/spark/operator/templates/spark-monitoring.yaml (hunk 2 above)
Comment (alembiewski) [reply repository/spark/operator/templates/spark-monitoring.yaml]: added, thanks!

**Comment by alembiewski** [comment]
Thanks for the review @akirillov! All comments have been addressed. Please take a look. 
Also, I just realize, that I made changes in the `beta1` docs folder instead of `latest`, fixed that as well.

File: repository/spark/docs/latest/monitoring.md [hunk 4]
```diff
@@ -98,14 +32,16 @@ the metrics on other one and see further instructions in next step.
          exposeExecutorMetrics: true
          prometheus:
            jmxExporterJar: "/prometheus/jmx_prometheus_javaagent-0.11.0.jar"
-           port: 8090
+           port: 8090
```
Comment (akirillov) [inline repository/spark/docs/latest/monitoring.md]: looks like an extra-newspace

File: repository/spark/docs/latest/monitoring.md [hunk 5]
```diff
@@ -98,14 +32,16 @@ the metrics on other one and see further instructions in next step.
          exposeExecutorMetrics: true
          prometheus:
            jmxExporterJar: "/prometheus/jmx_prometheus_javaagent-0.11.0.jar"
-           port: 8090
+           port: 8090 
      ```  
+     `spec.momitoring.prometheus.port` value should be the same for all submitted Spark Applications in order for metrics to be scraped.
```
Comment (akirillov) [inline repository/spark/docs/latest/monitoring.md:37]: 👍

**Review by akirillov — APPROVED** [review]
Thanks, @alembiewski. LGTM 👍

**Pull request merged**